bindata/assets/kube-descheduler/softtaintervalidatingadmissionpolicy.yaml

```diff
@@ -0,0 +1,52 @@
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "openshift-descheduler-softtainter-vap"
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    matchPolicy: Equivalent
+    namespaceSelector: {}
+    objectSelector: {}
+    resourceRules:
+      - apiGroups:   [""]
+        apiVersions: ["*"]
+        operations:  ["UPDATE"]
+        resources:   ["nodes"]
+        scope: "*"
+  matchConditions:
+    - name: 'openshift-descheduler-softtainter'
+      expression: "request.userInfo.username=='system:serviceaccount:openshift-kube-descheduler-operator:openshift-descheduler-softtainter' || request.userInfo.username=='system:serviceaccount:openshift-kube-descheduler-operator:openshift-descheduler'"
```

Comment:
Any reason why we also included the `openshift-descheduler` service account here ?

Reply:
With k8s RBAC rules we can only grant EDIT/PATCH rights at object level (nodes). We are using this `ValidatingAdmissionPolicy` to be more fine grained limiting it to soft-tatins.
This is needed for the service account used by the soft-tainter and also for the operator one since now it also needs node edit rights in order to be allowed to propagate them to the soft-tainter account that is dynamically created by the operator (only if the soft-tainter feature is enabled).
Descheduler service-account is not getting node edit rights so no need to limit it with the VAP.

Reply:
TIL.